salt/_pillar/metalk8s_nodes.py

```diff
@@ -156,21 +144,28 @@ def ext_pillar(minion_id, pillar, kubeconfig):
             if 'ca' in pillar['metalk8s']:
                 ca_minion = pillar['metalk8s']['ca'].get('minion', None)
 
-        client = kubernetes.config.new_client_from_config(
-            config_file=kubeconfig,
-        )
-
-        coreV1 = kubernetes.client.CoreV1Api(api_client=client)
-
-        node_list = coreV1.list_node()
-
-        cluster_version = get_cluster_version(api_client=client)
-        pillar_nodes = dict(
-            (node.metadata.name, node_info(node, ca_minion))
-            for node in node_list.items
-        )
-
-        volume_information = list_volumes(client, minion_id)
+        try:
+            node_list = __salt__['metalk8s_kubernetes.list_objects'](
+                kind='Node',
+                apiVersion='v1',
+                kubeconfig=kubeconfig
+            )
+        except Exception as exc:
+            log.exception(
+                "Failed to retrieve nodes for ext_pillar", exc_info=exc
+            )
+            pillar_nodes = __utils__['pillar_utils.errors_to_dict']([
+                "Failed to retrieve NodeList: {!s}".format(exc)
+            ])
+        else:
+            log.info("Successfully retrieved nodes for ext_pillar")
```

Comment:
I put this for debug purposes, maybe it should be removed or at least toned down to `debug` level